update readme for alpha release (#168)
ably-labs/models: README.md

  </a>
</p>

---
- [Overview](#overview)
- [Development Status](#development-status)
- [Quickstart](#quickstart)
  - [Prerequisites](#prerequisites)
  - [Installation and authentication](#installation-and-authentication)
  - [Instantiation](#instantiation)
  - [Creating a Model](#creating-a-model)
- [Documentation and examples](#documentation-and-examples)
- [Feedback](#feedback)

---

## Overview

> The Ably Models SDK is a key component of the [LiveSync](https://ably.com/docs/products/livesync) product that enables you to stream realtime updates from your database at scale to frontend clients.

Reviewer: ```suggestion
The Ably Models SDK is a key component of the [LiveSync](https://ably.com/docs/products/livesync) product that lets you stream realtime updates from your database at scale to frontend clients.
```

![LiveSync Diagram](/docs/images/what-is-livesync.png "LiveSync Diagram")

The Models SDK is a frontend library that simplifies subscribing to the changes in data models, applying optimistic updates and merging them with confirmed updates. It is a standalone SDK built on Ably’s JavaScript SDK with full TypeScript support.

The Database Connector and Ably Channels are the other two components of LiveSync that help publish changes from your database to frontend clients.

A model represents a data model of a specific part of your frontend application. Each frontend client can have multiple data models within the same application. 

![Models SDK Diagram](/docs/images/models-diagram.png "Models SDK Diagram")

When creating a new Model using the Models SDK you provide two functions to the Model a `sync()` function and a `merge()` function:
- The `sync(`) function is used by the SDK to retrieve the current state of the data model from your backend.
- The `merge()` function is used by the SDK to merge state change events published by the Database Connector with the existing frontend state in the Model.

> It is possible to use the Models SDK as a standalone library to merge new update events with an existing state, although we've optimized for it to be used as part of LiveSync.

Reviewer: ```suggestion
You can use the Models SDK as a standalone library to merge new update events with existing frontend state, but the SDK works best as part of LiveSync.
```

> The data models as part of the Models SDK remain synchronized with the realtime state of your database model. You can easily integrate this SDK into your project regardless of your frontend framework preference. You can take a look at the [examples](./examples) folder to see example usage in a React/NextJS application.

Reviewer: ```suggestion
The data models as part of the Models SDK remain synchronized with the state of your database, in realtime. You can easily integrate this SDK into your project regardless of which frontend framework you use.
```

## Development Status

> LiveSync, and hence the Models SDK, is in public alpha so that you can explore its capabilities. Your [feedback](https://docs.google.com/forms/d/e/1FAIpQLSd00n1uxgXWPGvMjKwMVL1UDhFKMeh3bSrP52j9AfXifoU-Pg/viewform) will help prioritize improvements and fixes for subsequent releases. The features in this release have been validated for a set of use-cases and the alpha is implemented to work under real-world situations and load, but you should use caution when deciding under what conditions to use it.

Reviewer: ```suggestion
LiveSync, and the Models SDK, is in public alpha so that you can explore its capabilities. Your [feedback](https://docs.google.com/forms/d/e/1FAIpQLSd00n1uxgXWPGvMjKwMVL1UDhFKMeh3bSrP52j9AfXifoU-Pg/viewform) will help prioritize improvements and fixes for later releases. The features in this release have been built to work under real-world situations and load, and for real-world use-cases, but there may still be some rough edges in this alpha.
```
